src/inversion_scripts/operators/TROPOMI_operator.py

```diff
@@ -457,76 +480,42 @@ def read_tropomi(filename):
 
     # Catch read errors in any of the variables
     try:
-        # Store methane, QA, lat, lon
-        tropomi_data = xr.open_dataset(filename, group="PRODUCT")
-        dat["methane"] = tropomi_data["methane_mixing_ratio_bias_corrected"].values[0, :, :]
-        dat["qa_value"] = tropomi_data["qa_value"].values[0, :, :]
-        dat["longitude"] = tropomi_data["longitude"].values[0, :, :]
-        dat["latitude"] = tropomi_data["latitude"].values[0, :, :]
-
-        # Store UTC time
-        reference_time = tropomi_data["time"].values
-        delta_time = tropomi_data["delta_time"][0].values
-        strdate = []
-        if delta_time.dtype == "<m8[ns]":
-            strdate = reference_time + delta_time
-        elif delta_time.dtype == "<M8[ns]":
-            strdate = delta_time
-        else:
-            print(delta_time.dtype)
-            pass
-        dat["utctime"] = strdate
-
-        # Store time for whole orbit
-        times = np.zeros(shape=dat["longitude"].shape, dtype="datetime64[ns]")
-        for k in range(dat["longitude"].shape[0]):
-            times[k, :] = strdate[k]
-        dat["time"] = times
-        tropomi_data.close()
+        # Store methane, QA, lat, lon, and time
+        with xr.open_dataset(filename, group="PRODUCT") as tropomi_data:
```

Comment:
Nice improvement by using `with`. Is the main benefit that it closes the stream automatically?